### PR TITLE
COMPOSER-1014: update ci runners

### PR DESCRIPTION
**Pull request by atodorov**
Cherry-picks commits from #1541 and resolves conflicts. 

Not all commits have been cherry-picked as of yet b/c they lead to lots of conflicts. I've taken the easiest ones first.

### CODE REVIEW

**Comment by Gundersanne** [comment]
To fix the fedora-33 failures https://github.com/osbuild/gitlab-ci-terraform/commit/bbbe80cf749033c4c6c67e912f5203a1e1b64bc5 can be used as a workaround.

File: schutzbot/mockbuild.sh [hunk 1]
```diff
@@ -88,12 +86,14 @@ greenprint "🧬 Using mock config: ${MOCK_CONFIG}"
...
-    greenprint "📋 Updating RHEL 8 mock template for unsubscribed image"
+if [[ "$ID" == rhel ]] && ! sudo subscription-manager status; then
+    greenprint "📋 Updating RHEL 8 mock template with the latest nightly repositories"
+    # strip everything after line with # repos
     sudo sed -i '/# repos/q' /etc/mock/templates/rhel-8.tpl
     # remove the subscription check
     sudo sed -i "s/config_opts\['redhat_subscription_required'\] = True/config_opts['redhat_subscription_required'] = False/" /etc/mock/templates/rhel-8.tpl
-    cat "$RHEL85_NIGHTLY_REPO" | sudo tee -a /etc/mock/templates/rhel-8.tpl > /dev/null
+    # reuse redhat.repo
+    cat /etc/yum.repos.d/rhel8internal.repo | sudo tee -a /etc/mock/templates/rhel-8.tpl > /dev/null
```
Comment (atodorov) [inline schutzbot/mockbuild.sh:96]: @ondrejbudai this fails on the rhel-9.0 runners b/c the filename is different, see https://gitlab.com/osbuild/ci/osbuild-composer/-/jobs/1608173404/. 

How do you want to handle this? Re-arrange the if conditions in this block or something else ?

File: test/cases/libvirt.sh [hunk 2]
```diff
@@ -2,19 +2,21 @@
...
 /usr/libexec/osbuild-composer-test/provision.sh
 
 # Test the images
 /usr/libexec/osbuild-composer-test/libvirt_test.sh qcow2
 
-/usr/libexec/osbuild-composer-test/libvirt_test.sh openstack
+#TODO: remove this condition once there is rhel9 support for openstack and vhd image types
+if [[ "$DISTRO_CODE" != rhel-90 ]]; then
+  /usr/libexec/osbuild-composer-test/libvirt_test.sh openstack
+fi
```
Comment (atodorov) [inline test/cases/libvirt.sh:16]: @ondrejbudai I'm not sure if this is like it is supposed to be or it's a mistake from resolving conflicts. Can you take a look?

File: test/cases/libvirt.sh [hunk 3]
```diff
@@ -2,19 +2,21 @@
...
 
 # RHEL 8.4 and Centos Stream 8 images also supports uefi, check that
-if [[ "$DISTRO_CODE" == "rhel_84" || "$DISTRO_CODE" == "rhel_85" || "$DISTRO_CODE" == "centos_8"  || "$DISTRO_CODE" == "rhel_90" ]]; then
+if [[ "$DISTRO_CODE" == "rhel-84" || "$DISTRO_CODE" == "centos-8" ]]; then
```
Comment (atodorov) [inline test/cases/libvirt.sh:19]: Hmm, feels like we should bring back the rest of the versions with a hyphen, no ?

File: test/data/repositories/rhel-8.json [hunk 4]
```diff
@@ -1,55 +1,66 @@
...
-    },
-    {
-      "name": "rt",
```
Comment (atodorov) [inline test/data/repositories/rhel-8.json:14]: This patch removes the definition for the RT repository. IIRC we still need this, right ?

File: test/data/repositories/rhel-84.json [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+      "rhsm": true,
+      "check_gpg": true
+    }
```
Comment (atodorov) [inline test/data/repositories/rhel-84.json:64]: Note: Missing definition for the RT repository.

File: schutzbot/mockbuild.sh (hunk 1 above)
Comment (ondrejbudai) [reply schutzbot/mockbuild.sh:96]: Run this block only for RHEL 8.

File: test/cases/libvirt.sh (hunk 2 above)
Comment (ondrejbudai) [reply test/cases/libvirt.sh:16]: No longer needed.

File: test/cases/libvirt.sh (hunk 3 above)
Comment (ondrejbudai) [reply test/cases/libvirt.sh:19]: yep

**Comment by atodorov** [comment]
@ondrejbudai only one of the regression tests is failing on 8.4 but passing everywhere else. Can we merge to avoid more delays? I am following up with @msehnout to figure out why we're seeing the regression failure.

File: schutzbot/mockbuild.sh [hunk 6]
```diff
@@ -87,12 +87,14 @@ greenprint "🧬 Using mock config: ${MOCK_CONFIG}"
...
-if [[ $VERSION_ID == 8.5 ]]; then
-    greenprint "📋 Updating RHEL 8 mock template for unsubscribed image"
+if [[ "$ID" == rhel && ${VERSION_ID%.*} == 8 ]] && ! sudo subscription-manager status; then
+    greenprint "📋 Updating RHEL 8 mock template with the latest nightly repositories"
+    # strip everything after line with # repos
     sudo sed -i '/# repos/q' /etc/mock/templates/rhel-8.tpl
     # remove the subscription check
     sudo sed -i "s/config_opts\['redhat_subscription_required'\] = True/config_opts['redhat_subscription_required'] = False/" /etc/mock/templates/rhel-8.tpl
-    cat "$RHEL85_NIGHTLY_REPO" | sudo tee -a /etc/mock/templates/rhel-8.tpl > /dev/null
+    # reuse redhat.repo
```
Comment (ondrejbudai) [inline schutzbot/mockbuild.sh:95]: Note to myself: I think we can remove RHEL85_NIGHTLY_REPO from secrets after this is merged.

**Review by ondrejbudai — APPROVED** [review]
Awesome! Thanks for pushing this over the finish line. :checkered_flag:

**Pull request merged**